src/test/java/org/prebid/server/bidder/emx_digital/EmxDigitalBidderTest.java

```diff
@@ -0,0 +1,415 @@
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("123", "2"))))
+				.build()))
+			.site(Site.builder().page("https://exmaple/").build())
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).hasSize(1)
+			.extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+			.flatExtracting(BidRequest::getImp)
+			.hasSize(1)
+			.first()
+			.satisfies(imp -> {
+				assertThat(imp.getTagid()).isEqualTo("123");
+				assertThat(imp.getSecure()).isEqualTo(1);
+				assertThat(imp.getBidfloor()).isEqualTo("2");
+				assertThat(imp.getBidfloorcur()).isEqualTo("USD");
+			});
+	}
+
+	@Test
+	public void makeHttpRequestsShouldModifyBannerFormatAndWAndHWhenRequestBannerWAndHIsNull() {
+		// given
+		final List<Format> formats = Arrays.asList(
+			Format.builder().h(20).w(21).build(),
+			Format.builder().h(30).w(31).build());
+
+		final BidRequest bidRequest = BidRequest.builder()
+			.imp(singletonList(Imp.builder()
+				.banner(Banner.builder().format(formats).build())
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("1", "asd"))))
+				.build()))
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		final Banner expectedBanner = Banner.builder().h(20).w(21)
+			.format(singletonList(Format.builder().h(30).w(31).build())).build();
+
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).hasSize(1)
+			.extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+			.flatExtracting(BidRequest::getImp).hasSize(1)
+			.first()
+			.satisfies(imp -> {
+				assertThat(imp.getTagid()).isEqualTo("1");
+				assertThat(imp.getSecure()).isEqualTo(0);
+				assertThat(imp.getBidfloor()).isNull();
+				assertThat(imp.getBidfloorcur()).isNull();
+				assertThat(imp.getBanner()).isEqualTo(expectedBanner);
+			});
+	}
+
+	@Test
+	public void makeHttpRequestsShouldSendRequestToModifiedUrlWithHeaders() {
+		// given
+		final BidRequest bidRequest = BidRequest.builder()
+			.imp(singletonList(Imp.builder()
+				.banner(Banner.builder().w(1).h(1).build())
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("1", "asd"))))
+				.build()))
+			.device(Device.builder().ip("ip").ua("Agent").language("fr").dnt(1).build())
+			.site(Site.builder().page("myPage").build())
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).hasSize(1)
+			.first()
+			.satisfies(request ->
+				assertThat(request.getUri()).startsWith("https://test.endpoint.com?t=1000&ts="));
```

Comment:
Now here a `.satisfies` verification can be useful - you can add another sub check that would take a sub-string from 
`"ts="` to string size and make an int out of it, take a current time and make an int value `test time` in `//given`, and then assert that one is bigger than other.

Also, please add a small test case when we are in testing (constant url).